src/index.js

```diff
@@ -89,15 +103,17 @@ class IPLDResolver {
       throw new Error('`path` argument must be a string')
     }
 
+    const ipld = this
+
     const generator = async function * () {
-      // End iteration if there isn't a CID to follow anymore
+      // End iteration if there isn't a CID to follow any more
       while (cid !== null) {
```

Comment:
Should this be an endless loop instead for clarrity? Now that we don't actually stop the loop due to that condition, but at the bottom?

Reply:
Or perhaps doing a `while (CID.isCIDC(cid))`?

Reply:
`while (CID.isCIDC(cid))` will never evaluate to false because we `return` before that would happen.

> Should this be an endless loop instead for clarity?

Makes sense, I'll update the PR